java/arcs/jvm/host/Util.kt

```diff
@@ -0,0 +1,23 @@
+package arcs.jvm.host
+
+import arcs.core.host.Particle
+import arcs.core.host.ParticleConstructor
+import arcs.core.host.ParticleIdentifier
+import arcs.core.host.ParticleRegistration
+import arcs.core.host.toParticleIdentifier
+import kotlin.reflect.KClass
+
+/** Returns a pair mapping [ParticleIdentifier] to [ParticleConstructor] */
+fun KClass<out Particle>.toRegistration(): Pair<ParticleIdentifier, ParticleConstructor> =
+    this.toParticleIdentifier() to this.toConstructor()
+
+/** Return a [ParticleConstructor] via Kotlin reflection. */
+fun KClass<out Particle>.toConstructor(): ParticleConstructor = {
+    this@toConstructor.java.newInstance()
```

Comment:
Not sure I'm a fan of using reflection to instantiate the particles..

Reply:
maybe ctor reference instead? e.g. ReadingHost(::ReadPerson) instead of ReadingHost(ReadPerson::class) ?

Reply:
If that works, definitely!

Reply:
inline reified functions to the rescue!